guiclient/unpostedPoReceipts.cpp

```
@@ -196,74 +196,218 @@ void unpostedPoReceipts::sPost()
+                    "  AND isControlledItemsite(itemsite_id);");
+        tohead.bindValue(":toitem_id", recvInfo.value("recv_orderitem_id").toInt());
+        tohead.exec();
+        if (tohead.first())
+        {
+          parentItemlocdist.prepare("SELECT createItemlocdistParent(:itemsite_id, :qty, :orderType, :orderitemId, "
+           ":itemlocSeries, NULL, NULL, 'TW') AS result;");
+          parentItemlocdist.bindValue(":itemsite_id", tohead.value("itemsite_id").toInt());
+          parentItemlocdist.bindValue(":qty", recvInfo.value("recv_qty").toDouble() * -1);
+          parentItemlocdist.bindValue(":orderType", recvInfo.value("recv_order_type").toString());
+          parentItemlocdist.bindValue(":orderitemId", recvInfo.value("recv_orderitem_id").toInt());
+          parentItemlocdist.bindValue(":itemlocSeries", itemlocSeries);
+          parentItemlocdist.exec();
+          if (parentItemlocdist.first())
+          {
+            // If the "to" itemsite is not controlled, distribute the From side here
+            if (!recvInfo.value("controlled").toBool())
+            {
+              if (distributeInventory::SeriesAdjust(itemlocSeries, this, QString(), QDate(), QDate(),
+                true) == XDialog::Rejected)
+              {
+                cleanup.exec();
+                QMessageBox::information( this, tr("Enter Receipts"),
+                  tr("Posting Distribution Detail Failed for 'From' itemsite") );
+                return;
+              }
+            }
+            else 
+              itemlocdistId = parentItemlocdist.value("result").toInt();
+          }
+          else 
+          {
+           cleanup.exec();
+           ErrorReporter::error(QtCriticalMsg, this, tr("Error Creating itemlocdist Record for "
+             "'From' itemsite"), parentItemlocdist, __FILE__, __LINE__);
+           return;
+          }
+        }
+        else if (tohead.lastError().type() != QSqlError::NoError)
```

Should throw an error here instead of just returning.